File: internal/content/testdata/bench/javascript/nc-container-most-water.js
function max_area(heights) {
    let i = 0, j = heights.length - 1;
    let best = 0;
    while (i < j) {
        const area = Math.min(heights[i], heights[j]) * (j - i);
        best = Math.max(best, area);
        if (heights[i] < heights[j]) i++;
        else j--;
    }
    return best;
}

File: internal/content/testdata/bench/go/nc-container-most-water.go
func max_area(heights []int) int {
	i, j := 0, len(heights)-1
	best := 0
	for i < j {
		h := heights[i]
		if heights[j] < h {
			h = heights[j]
		}
		area := h * (j - i)
		if area > best {
			best = area
		}
		if heights[i] < heights[j] {
			i++
		} else {
			j--
		}
	}
	return best
}
